src/pages/index.js

```diff
@@ -6,159 +7,235 @@ import PopupWithForm from "../components/PopupWithForm.js";
-  "#edit-modal",
-  handleProfileEditFormSubmit
-);
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      job: res.about,
+      avatar: res.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+// Image Popup
 
 const popupImage = new PopupWithImage({
   popupSelector: "#image-modal",
 });
 
-newCardPopup.setEventListeners();
-profileEditPopup.setEventListeners();
 popupImage.setEventListeners();
 
-profileEditCloseButton.addEventListener("click", () => {
-  profileEditPopup.close();
-});
-
 // Profile Edit Modal
-function openProfileEditForm() {
+
+const profileEditPopup = new PopupWithForm(
+  "#edit-modal",
+  handleProfileEditFormSubmit
+);
+
+profileEditPopup.setEventListeners();
+
+function fillProfileForm() {
   const profileInfo = userInfo.getUserInfo();
   profileTitleInput.value = profileInfo.name;
   profileDescriptionInput.value = profileInfo.job;
-  profileEditPopup.open();
 }
 
-profileEditButton.addEventListener("click", openProfileEditForm);
+profileEditButton.addEventListener("click", () => {
+  fillProfileForm();
+  profileEditPopup.open();
+});
+
+function handleProfileEditFormSubmit(profileInfo) {
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(profileInfo)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        job: res.about,
+      });
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => profileEditPopup.setLoading(false));
 
-function handleProfileEditFormSubmit(inputValues) {
-  const profileInfo = {};
-  profileInfo.name = inputValues.title;
-  profileInfo.job = inputValues.description;
-  console.log(profileInfo);
-  userInfo.setUserInfo(profileInfo);
   profileEditPopup.close();
 }
 
-// Initial Cards
-function handleImageClick(cardData) {
-  popupImage.open(cardData);
+profileEditCloseButton.addEventListener("click", () => {
+  profileEditPopup.close();
+});
+
+// Change Avatar Modal
+
+const changeImageForm = new PopupWithForm(
+  "#change-avatar-modal",
+  handleChangeAvatarFormSubmit
+);
+changeImageForm.setEventListeners();
+
+const changeAvatarButton = document.querySelector(
+  ".change-avatar__edit-button"
+);
+
+changeAvatarButton.addEventListener("click", (evt) => {
+  evt.preventDefault;
+  changeImageForm.open();
+});
+
+function handleChangeAvatarFormSubmit(profileInfo) {
+  changeImageForm.setLoading(true);
+  api
+    .updateAvatar(profileInfo)
+    .then((res) => {
+      userInfo.setAvatar({ avatar: res.avatar });
+      changeImageForm.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => changeImageForm.setLoading(false));
+}
+
+// Delete Card
+
+const deleteImageModal = document.querySelector("#delete-image-modal");
+const deleteImageConfirmationButton =
+  deleteImageModal.querySelector(".modal__button");
+
+deleteImageConfirmationButton.addEventListener("click", () => {
+  deleteConfirmation.setLoading(true);
```

Comment:
Please, delete it

`deleteConfirmation.setLoading(true);` should be done before the request (not after clicking)

It should be done in line 154